aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -539,18 +550,23 @@ std::tuple<Tensor&, Tensor&> min_out(
   return result;
 }
 
-Tensor& clamp_out(const Tensor& self, const c10::optional<Scalar>& min, const c10::optional<Scalar>& max, Tensor& result) {
+TORCH_IMPL_FUNC(clamp_out)
+(
+ const Tensor& self,
+ const OptionalScalarRef min,
+ const OptionalScalarRef max,
+ const Tensor& result) {
   if (min && max) {
-    auto iter = TensorIterator::unary_op(result, self);
-    clamp_scalar_stub(iter.device_type(), iter, *min, *max);
+    clamp_scalar_stub(device_type(), *this, min.toScalar(), max.toScalar());
   } else if (max) {
-    at::clamp_max_outf(self, *max, result);
+    Tensor temp(result);
+    at::clamp_max_outf(self, max.toScalar(), temp);
+    result.copy_(temp);
```

Comment:
Oh, this is another snippet I wanted to talk about. I would like to pass in `result` directly, but can't do that since the last parameter of `at::clamp_max_outf` is not `const`. Can this be fixed once `clamp_max` is ported to structured? Should I stack that + the fix for this on top?

Reply:
This kind of looks like #50953.

Reply:
Do a const cast

Reply:
(Don't `Tensor temp(result)`, that's an unnecessary refcount bump). In the terminal state `clamp_min_outf` takes a const argument and the const cast is unnecessary)